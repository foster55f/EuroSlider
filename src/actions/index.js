export const addGames = games => ({
    type: 'ADD_GAMES',
    games
})

export const filterGames = displayGames => ({
    type: 'FILTER_GAMES',
    displayGames
})

