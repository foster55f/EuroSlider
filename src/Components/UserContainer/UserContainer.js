import React from 'react'
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { userLogOut} from '../../actions';





class UserContainer extends React.Component {
    constructor() {
        super();
        this.state = {

        }
    }   

    logOut = () => {
        this.props.userLogOut()
        this.props.history.push('/')
        
    }

    render() {
        return (
            <>
                <h1>{`Welcome ${this.props.user.name}`}</h1>
                <button onClick ={this.logOut}>Log Out</button>
            </>
        )
    }
}
export const mapStateToProps = state => ({
    user: state.user,
    userLog: state.userLog
})

export const mapDispatchToProps = dispatch => ({
    userLogOut: loggedIn => dispatch(userLogOut(loggedIn))
})

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(UserContainer))
