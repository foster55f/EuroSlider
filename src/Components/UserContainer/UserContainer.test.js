import React from 'react';
import { shallow } from 'enzyme';
import { UserContainer, mapStateToProps, mapDispatchToProps } from './UserContainer';
import { addUser, userLogOut } from '../../actions';


describe('UserContainer', () => {
  let wrapper;

  it('should match the App snapshot', () => {
    let wrapper = shallow(<UserContainer />);
    expect(wrapper).toMatchSnapshot();
  })


  describe('mapsStateToProps', () => {
    it('should return only movies and loading properties from the store', () => {
        const mockState = {
            user: {},
            userLog: false,
        };
        const expected = {
            user: {},
            userLog: false
        };
        const mappedProps = mapStateToProps(mockState);

        expect(mappedProps).toEqual(expected);
    });
});

describe('mapDispatchToProps', () => {
    it('calls dispatch with an addMovies action when addMovies is called',
        () => {
            const mockDispatch = jest.fn();
            const actionToDispatch = userLogOut(false);
            const mappedProps = mapDispatchToProps(mockDispatch);

            mappedProps.userLogOut(false);

            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });  
})