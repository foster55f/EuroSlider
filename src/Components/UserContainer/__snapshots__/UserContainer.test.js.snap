// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserContainer should match the App snapshot 1`] = `
<Fragment>
  <h1>
    Welcome 
  </h1>
  <button
    onClick={[Function]}
  >
    Log Out
  </button>
</Fragment>
`;
